.circleci/config.yml

```diff
@@ -0,0 +1,86 @@
+version: 2.1
+
+commands:
+  run_with_pip_cache:
+    description: Run the passed steps with a restored pip cache
+    parameters:
+      cache_key_base:
+        type: string
+        description: Base part of the cache key. Version and a more specific checksum of files will be added too.
+      cache_key_version:
+        type: string
+        description: Version prefix. Can be changed to invalidate caches.
+        default: v1
+      steps:
+        type: steps
+    steps:
+      - run: cat tox.ini dev-constraints.txt > check.txt
+      - restore_cache:
+          name: Restoring pip wheel cache
+          keys:
+            - &key_full << parameters.cache_key_version >>-<< parameters.cache_key_base >>-{{ checksum "check.txt" }}
+            - &key << parameters.cache_key_version >>-<< parameters.cache_key_base >>
+      - steps: << parameters.steps >>
+      - save_cache:
+          name: Saving pip wheel cache
+          key: *key
+          paths:
+            - ~/.cache/pip
+      - save_cache:
+          name: Saving pip wheel cache
```

Comment:
nit: should this have a different name to distinguish it from the other save_cache above?